File: minecraft-web-manager/backend/src/api/controllers/system.controller.js
import systemService from '../../services/system.service.js';

class SystemController {
  /**
   * GET /system/info
   * Obtener información del sistema
   */
  static async getSystemInfo(req, res) {
    try {
      const info = await systemService.getSystemInfo();

      return res.json({
        success: true,
        data: info
      });
    } catch (error) {
      return res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  /**
   * GET /system/logs
   * Obtener logs del sistema
   */
  static async getLogs(req, res) {
    try {
      const { limit = 50, offset = 0, action, userId } = req.query;

      let logs;

      if (action) {
        logs = await systemService.getLogsByAction(action, parseInt(limit));
      } else if (userId) {
        logs = await systemService.getLogsByUser(parseInt(userId), parseInt(limit));
      } else {
        logs = await systemService.getLogs(parseInt(limit), parseInt(offset));
      }

      return res.json({
        success: true,
        data: logs
      });
    } catch (error) {
      return res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }
}

export default SystemController;

File: minecraft-web-manager/backend-python/app/api/controllers/system_controller.py
from fastapi import HTTPException, status

from app.services.system_service import SystemService


class SystemController:
    def __init__(self):
        self.system_service = SystemService()
    
    async def get_info(self):
        """Obtener información del sistema"""
        try:
            info = await self.system_service.get_system_info()
            return info
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al obtener información del sistema: {str(e)}"
            )
    
    async def health_check(self):
        """Health check del sistema"""
        return {
            "status": "ok",
            "service": "minecraft-web-manager",
            "version": "2.0.0"
        }
